### PR TITLE
Referrals: Claim pass

### PR DESCRIPTION
**Pull request by SergioEstevao**
| 📘 Part of: #2083  |  

This PR implements only the UI for the claim passes flow. The signup and IAP flows will be done on a separate PR
At the moment because we don't have any server control logic the banner to claim passes will always be present .

| 1 | 2 |
| - | - |
| Simulator Screenshot - iPhone 15 Pro - 2024-09-11 at 00 03 52 | Simulator Screenshot - iPhone 15 Pro - 2024-09-11 at 00 04 01 | 
| Simulator Screenshot - iPad mini (6th generation) - 2024-09-11 at 00 09 21 | Simulator Screenshot - iPad mini (6th generation) - 2024-09-11 at 00 09 25 |

## To test

1. Start the app
2. Ensure that you have the referrals FF enabled
3. Open profile
4. Check that you see the banner for claim passes
5. Tap on it
6. See that you see the Claim pass view in a sheet
7. Tap on not now
8. Check that you are returned to Profile

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/59">7.73</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: podcasts/Referrals/Banner/ReferralsClaimBannerTableCell.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        let viewModel = ReferralClaimPassModel()
+
+        let bannerView = ReferralsClaimBannerView(viewModel: viewModel).themedUIView
+        bannerView.backgroundColor = .clear
+        contentView.addSubview(bannerView)
+
+        bannerView.translatesAutoresizingMaskIntoConstraints = false
+        NSLayoutConstraint.activate([
+            bannerView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: 16.0),
+            bannerView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -16.0),
```
Comment (bjtitus) [inline podcasts/Referrals/Banner/ReferralsClaimBannerTableCell.swift]: Could these `16` values go in a Constants Enum for horizontal spacing?

**Review by bjtitus — APPROVED** [review]
👍 One minor nit but this looked good in testing.

File: podcasts/Referrals/Banner/ReferralsClaimBannerTableCell.swift (hunk 1 above)
Comment (SergioEstevao) [reply podcasts/Referrals/Banner/ReferralsClaimBannerTableCell.swift]: Done here: https://github.com/Automattic/pocket-casts-ios/pull/2158/commits/2f6f95b56e195054af066036208a783096f89ef0

**Pull request merged**